android/java/org/chromium/chrome/browser/crypto_wallet/fragments/NftGridFragment.java

```diff
@@ -0,0 +1,332 @@
+import org.chromium.chrome.browser.app.helpers.Api33AndPlusBackPressHelper;
+import org.chromium.chrome.browser.crypto_wallet.BlockchainRegistryFactory;
+import org.chromium.chrome.browser.crypto_wallet.activities.BraveWalletActivity;
+import org.chromium.chrome.browser.crypto_wallet.activities.NftDetailActivity;
+import org.chromium.chrome.browser.crypto_wallet.adapters.WalletCoinAdapter;
+import org.chromium.chrome.browser.crypto_wallet.adapters.WalletNftAdapter;
+import org.chromium.chrome.browser.crypto_wallet.listeners.OnWalletListItemClick;
+import org.chromium.chrome.browser.crypto_wallet.model.WalletListItemModel;
+import org.chromium.chrome.browser.crypto_wallet.util.AssetUtils;
+import org.chromium.chrome.browser.crypto_wallet.util.JavaUtils;
+import org.chromium.chrome.browser.crypto_wallet.util.PortfolioHelper;
+import org.chromium.chrome.browser.crypto_wallet.util.Utils;
+import org.chromium.chrome.browser.crypto_wallet.util.WalletUtils;
+import org.chromium.chrome.browser.custom_layout.AutoFitVerticalGridLayoutManager;
+import org.chromium.chrome.browser.util.LiveDataUtil;
+import org.chromium.content_public.browser.UiThreadTaskTraits;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class NftGridFragment extends Fragment implements OnWalletListItemClick {
+    private static final String TAG = "NftGridFragment";
+    private static final float NFT_ITEM_WIDTH_DP = 180;
+
+    private WalletModel mWalletModel;
+    private PortfolioModel mPortfolioModel;
+    private List<PortfolioModel.NftDataModel> mNftDataModels;
+    private List<NetworkInfo> mAllNetworkInfos;
+
+    private PortfolioHelper mPortfolioHelper;
+    private NetworkInfo mNetworkInfo;
+
+    private RecyclerView mRvNft;
+    private WalletNftAdapter mWalletNftAdapter;
+    private ProgressBar mPbAssetDiscovery;
+    private ViewGroup mAddNftsContainer;
+
+    public static NftGridFragment newInstance() {
+        return new NftGridFragment();
+    }
+
+    private JsonRpcService getJsonRpcService() {
+        Activity activity = getActivity();
+        if (activity instanceof BraveWalletActivity) {
+            return ((BraveWalletActivity) activity).getJsonRpcService();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            Api33AndPlusBackPressHelper.create(
+                    this, (FragmentActivity) requireActivity(), () -> requireActivity().finish());
+        }
+        try {
+            BraveActivity activity = BraveActivity.getBraveActivity();
+            mWalletModel = activity.getWalletModel();
+            mPortfolioModel = mWalletModel.getCryptoModel().getPortfolioModel();
+            mWalletModel.getCryptoModel().getPortfolioModel().discoverAssetsOnAllSupportedChains();
```

Comment:
`discoverAssetsOnAllSupportedChains` is required? this should be called in the `PortfolioFragment` so should be enough to be called once. Unless not being called inside `PortfolioFragment` with this PR?

Reply:
This is required in case the activity is restored from background and the UI is landing directly in the NFTs section. Without this call the user should always scroll back to the Portfolio section to trigger the discovery of the assets.

Reply:
We don't have to worry about that case either. This should be called after wallet login to discover assets, do we need to call this again?

Reply:
Okay, then there's no way to reach this UI without triggering the asset discovery. I'm removing this and reverting the other bits connected to it.